src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,141 @@
+
+    fs - The FilterSet to which filters will be appended.
+    filter_query - The filter query parameters containing field names and filter strings.
+    columns - The columns to which the filters apply, represented as a dictionary
+        mapping field names to AttributesColumn objects.
+    trans_field_func - A function to transform the field name to valid column name in the filter query
+        before applying it to the FilterSet. Default is None.
+
+    """
+    for field, querystring in filter_query.items():
+        if field not in columns:
+            raise errors.IllegalParameterError(f"No such filter field: {field}")
+        column = columns[field]
+        minlen = analyzers.get_minimum_query_length(column.filter_strategy)
+        if minlen and len(querystring) < minlen:
+            raise errors.IllegalParameterError(
+                f"Filter field '{field}' requires a minimum query length of {minlen}")
+
+        # Translates the query field name to an available column name if necessary.
+        # For example, translates query field name '1' to a valid column name 'col_1_val'.
+        field = trans_field_func(field) if trans_field_func else field
+
+        fs.append(
+            field,
+            column.type,
+            querystring,
+            analyzers.get_analyzer(column.filter_strategy),
+            column.filter_strategy
+        )
+    return fs
+
+
+async def get_filters(
+        r: Request,
+        arango_coll: str,
+        coll_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        data_product: str,
+        get_columns_func: callable,
+        view_name: str = None,
+        count: bool = False,
+        sort_on: str = None,
+        sort_desc: bool = False,
+        filter_conjunction: bool = True,
+        match_spec: SubsetSpecification = None,
+        selection_spec: SubsetSpecification = None,
+        keep: dict[str, set[col_models.ColumnType]] = None,
+        keep_filter_nulls: bool = False,
+        skip: int = 0,
+        limit: int = 1000,
+        start_after: str = None,
+        trans_field_func: callable = None,
+) -> FilterSet:
+    """
+    Constructs a FilterSet and applies filters using the provided filter query and columns.
+
+    Requires a 'get_columns_func' to generate columns, represented as a dictionary mapping column name/ID to
+    columnar_attribs_common_models.AttributesColumn objects.
+    """
+    filter_query = _get_filter_map(r)
+    if filter_query and not view_name:
+        if load_ver_override:
+            # If we need this feature than the admin needs to supply the view name to use
+            # via the API
+            raise ValueError("Filtering is not supported with a load version override.")
+        raise ValueError(f"No search view name configured for collection {coll_id}, "
+                         + f"data product {data_product}. Cannot perform filtering operation")
```

Comment:
This is assuming a manually configured view name. For the heatmaps, it's always possible to get the view name, not not necessarily true that the view exists, unlike manually configured views (at least if they haven't been manually deleted). How are you planning to handle this with heatmaps?

Reply:
For heatmaps, this will never be triggered. Either manually configured view and generic view are guaranteed to exist in the DB. We can just assume at this step, the view does exist in the DB and if that's not the case, the query will error.

Reply:
>  Either manually configured view and generic view are guaranteed to exist in the DB

A generic view won't exist in the DB if the service manager hasn't been run

Reply:
No, because the service checks that the view exists when registering a Collection, so if the view_name != None there's a guaranteed view. There's no equivalent for generic views.

Reply:
unless you want to explicitly check the view existence here or somewhere which I really don't think it's necessarily. I think it's just fine to have the query throws an error in that situation. The error msg raised by AQLQuery is pretty clear what's going on. 

`
(dev) tgu@collections (dev_heatmap_filtering)$curl -X 'GET' \
  'http://127.0.0.1:5000/collections/PMI/data_products/microtrait/?limit=1&filter_49=false' \
  -H 'accept: application/json' \
  -H "Authorization: Bearer $token" | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   227  100   227    0     0    542      0 --:--:-- --:--:-- --:--:--   545
{
    "error": {
        "httpcode": 500,
        "httpstatus": "Internal Server Error",
        "time": "2023-12-19T23:54:52.265016+00:00",
        "message": "[HTTP 404][ERR 1203] AQL: collection or view not found: name: microtrait_generic_view_missing (while parsing)"
    }
}
`

Reply:
I mean it's just
```
if not storage.has_view(view_name):
    throw ValueError(...)
```
So the user gets a slightly less gross error

Reply:
👍

Reply:
:+1: